File: src/main/resources/output/IFC4X3_python/Entities/IfcAlignmentParameterSegment/IfcAlignmentHorizontalSegment/IfcAlignmentHorizontalSegment.py
class IfcAlignmentHorizontalSegment(IfcAlignmentParameterSegment):
    def __init__(self):
        self.StartPoint: IfcCartesianPoint = None
        self.StartDirection: IfcPlaneAngleMeasure = None
        self.StartRadiusOfCurvature: IfcLengthMeasure = None
        self.EndRadiusOfCurvature: IfcLengthMeasure = None
        self.SegmentLength: IfcNonNegativeLengthMeasure = None
        self.GravityCenterLineHeight: IfcPositiveLengthMeasure = None
        self.PredefinedType: IfcAlignmentHorizontalSegmentTypeEnum = None

File: output/ifc4x3_dev_73740fe4/entity/IfcAlignmentHorizontalSegment.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


public class IfcAlignmentHorizontalSegment extends IfcAlignmentParameterSegment {

    public IfcCartesianPoint StartPoint;
    public IfcPlaneAngleMeasure StartDirection;
    public IfcLengthMeasure StartRadiusOfCurvature;
    public IfcLengthMeasure EndRadiusOfCurvature;
    public IfcNonNegativeLengthMeasure SegmentLength;
    public IfcPositiveLengthMeasure GravityCenterLineHeight; // OPTIONAL
    public IfcAlignmentHorizontalSegmentTypeEnum PredefinedType;
}
